docs/diagrams/UndoRedo0.puml

```diff
@@ -0,0 +1,39 @@
+@startuml
+hide circle
+skinparam classAttributeIconSize 0
+
+Command <|-- RedoableCommand
+StackUndoRedo *--> "2" Stack
+RedoableCommand "*" <-left-  Stack
+class Command {
+  # undoRedoStack: StackUndoRedo
+  + execute(): CommandResult
+  + setData(): void
+}
+
+
+
+class RedoableCommand
+ {
+   - previousAddressBook: ReadOnlyAddressBook
+   - previousArchiveBook: ReadOnlyAddressBook
+   - isSwapped: boolean
+   - saveAddressBookSnapshot(): void
```

Comment:
Maybe we should start updating the RedoableCommand to save the state of the `ReadOnlyShop` instead?
Will probably need the code to reflect this as well in `RedoableCommand` class.